### PR TITLE
Allow extend Backend menu

### PR DESCRIPTION
**Pull request by versh23**

### CODE REVIEW

**Comment by bobdenotter** [comment]
@versh23 Running `vendor/bin/ecs check src` and `vendor/bin/phpstan src` should give more details about the Code Style notices. :-)

**Comment by versh23** [comment]
@bobdenotter sorry :) make install failed with composer v2 and hirak/prestissimo. now fix it and run fixers

**Comment by bobdenotter** [comment]
No need to apologize, mate! We're more than happy to explain about common pitfalls 👍

File: src/Menu/BackendMenuBuilder.php [hunk 1]
```diff
@@ -37,8 +37,12 @@ final class BackendMenuBuilder implements BackendMenuBuilderInterface
     /** @var ContentExtension */
     private $contentExtension;
 
+    /** @var ExtensionBackendMenuInterface[] */
+    private $extensionMenu;
+
     public function __construct(
         FactoryInterface $menuFactory,
+        iterable $extensionMenu,
```
Comment (simongroenewolt) [inline src/Menu/BackendMenuBuilder.php:45]: maybe use plural $extensionMenus to indicate it is a list?

**Review by simongroenewolt — COMMENTED** [review]
Looks good! One thing I'm thinking about is if it is sufficient to put menu-items after the list of content-types and before the config menu header.
I think this means you need one extension to insert some sort of an additional header in the menu, or are they all required/advised to add their own heading? Otherwise it might look like the items belong to the content-types list. (Sorry, didn't install to test this myself, maybe there already is a clear division)
Also: should only one location be available? Could it make sense to allow to add an extension item to the settings menu, or somewhere else, all the way at the top maybe?

File: src/Menu/BackendMenuBuilder.php (hunk 1 above)
Comment (versh23) [reply src/Menu/BackendMenuBuilder.php]: correct!

**Comment by versh23** [comment]
@simongroenewolt 
example
`class ExtensionMenu implements ExtensionBackendMenuInterface
{

    public function addItems(MenuItem $menu): void
    {
        $menu->addChild('Billing extension', [
            'extras' => [
                'name' => 'Billing extension',
                'type' => 'separator',
                'icon' => 'fa-wrench',
            ],
        ]);

        $menu->addChild('Billing settings', [
            'uri' => '/bolt/test',
            'extras' => [
                'name' => 'Billing settings',
                'icon' => 'fa-sliders-h',
            ],
        ]);
    }
}
`

image


looks good for me.
may be set name of extension to separator automaticaly or add method in interface for it ( menuName\header or else)

about sorting.. hmm.. may be refactor BackendMenuBuilder and separate menu parts put in ExtensionBackendMenuInterface (Content, Settings) and now we can sort it by priority. then we need rename ExtensionBackendMenuInterface -> BackendMenuInterface I guess


What do you think?

**Review by bobdenotter — APPROVED** [review]
I'm really liking this! Nice and clean.. 

We'll need to be sure to write documentation for this, when it lands in Bolt 4.2 :-)

File: .github/workflows/code_analysis.yaml [hunk 2]
```diff
@@ -27,7 +27,7 @@ jobs:
 
                     -
                         name: Check YAML configs files
-                        run: bin/console lint:yaml config --ansi
+                        run: bin/console lint:yaml config --ansi --parse-tags
```
Comment (bobdenotter) [inline .github/workflows/code_analysis.yaml:30]: Is this part of this PR?

Comment (versh23) [reply .github/workflows/code_analysis.yaml:30]: yes, see https://github.com/symfony/symfony/issues/28465

**Comment by I-Valchev** [comment]
Very cool! 👍

**Comment by versh23** [comment]
no need to order items in backend menu?

**Pull request merged**